CMakeLists.txt

```diff
@@ -507,9 +498,21 @@ SET(CMAKE_REQUIRED_LIBRARIES
 
 IF(MPI_FOUND)
   # Mumps
-  IF(WITH_Mumps)
-    FIND_PACKAGE(Mumps REQUIRED)
-  ENDIF()
+  if(WITH_Mumps)
+    find_package(MUMPS CONFIG)
+    if(MUMPS_FOUND)
+      set(Mumps_FOUND true)
+      set(Mumps_LIBRARIES ${MUMPS_LIBRARIES})
+      set(Mumps_INCLUDE_DIR ${MUMPS_INCLUDE_DIRS})
+      if(MUMPS_METIS_FOUND)
+        set(Metis_FOUND true)
+        set(Metis_LIBRARIES ${METIS_LIBRARIES})
+        set(Metis_INCLUDE_DIR ${METIS_INCLUDE_DIRS})
+      endif()
+    else()
+      FIND_PACKAGE(Mumps REQUIRED)
+    endif()
+  endif()
```

Comment:
As background for why moving that part to `FindMumps.cmake` would be nice:
Currently, ElmerFEM doesn't install .cmake files. (It only optionally installs pkg-config files.) But if at some point in the future a feature that installs cmake modules alongside ElmerFEM should be integrated, that hunk of logic would need to be repeated wherever those installed .cmake modules would be used in downstream code.
It would be a lot cleaner if that hunk of logic would be integrated in the existing `FindMUMPS.cmake` file.